### ISSUE TITLE
Server-side rendering can cause style[amp-custom] CSS overage

### ISSUE THREAD

**Issue by westonruter** [ampproject/amp-wp#5517]
## Bug Description

Given an AMP page that has an `style[amp-custom]` which is just under the 75000 byte limit at 74,998 bytes. 

If this AMP page also has an `amp-analytics` tag, then the SSR transformer will end up adding an inline `style` to this element like so:

```html
<amp-analytics type="analytics" class="i-amphtml-layout-fixed i-amphtml-layout-size-defined" style="width:1px;height:1px;" i-amphtml-layout="fixed">
```

The result is an AMP validation error is leaked to the frontend:

> The author stylesheet specified in tag ‘style amp-custom’ and the combined inline styles is too large – document contains 75023 bytes whereas the limit is 75000 bytes.

Why is this? It's because `width:1px;height:1px;` (21 bytes) is _added_ to the size of `style[amp-custom]` (74,998 bytes): 74,998 + 21 = 75,023.

Originally reported in support topic: https://wordpress.org/support/topic/gcs-error/

## Expected Behaviour

SSR should not cause the total `amp-custom` CSS to go over 75000 bytes. It should keep track of how much CSS it is adding to the page, and if that causes the total to go over what is already in `style[amp-custom]`, it should abort SSR.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

**Comment by schlessera** [comment]
I'm surprised by this, as the `ServerSideRendering` transformer keeps track of the byte count of `style[@amp-custom]` specifically to avoid this situation.

I'll have to dig deeper to see why this is currently failing.

**Comment by schlessera** [comment]
I cannot replicate this so far. Even when I copy the exact HTML from the page that had the validation issue and run it through the optimizer, I have the end result of the optimizer still stay at 74998 bytes, as it checks the accumulated size before adding CSS.

**Comment by schlessera** [comment]
I've added a draft PR that adds a test to show that the SSR correctly stays below the maximum bye count: https://github.com/ampproject/amp-wp/pull/5521/files#diff-76459e2d3e79daf78403e793d6f441ecacd167962a913a42973126f4310c526eR311-R327

**Comment by schlessera** [comment]
Oh, I see the issue now. The validator counts the inline styles from the `i-amphtml-*` elements as well. Wouldn't that be a bug in the validator? These are supposed to be what the runtime would add anyway, only ... rendered on the server?

**Comment by schlessera** [comment]
It looks like this would be broken in the Go & the Node.JS versions as well:
* Go: https://github.com/ampproject/amppackager/blob/ca6f35a1b2e65f02f8e55c9e9bc218f25b801dc8/transformer/layout/layout.go#L197
* Node.JS: https://github.com/ampproject/amp-toolbox/blob/72ac57e0b3d1a57fff3a083abc8b2f6aef7babb3/packages/optimizer/lib/transformers/ApplyLayout.js#L94

cc @sebastianbenz

**Comment by westonruter** [comment]
Relates to https://github.com/ampproject/amphtml/issues/27468

**Comment by westonruter** [comment]
> Oh, I see the issue now. The validator counts the inline styles from the `i-amphtml-*` elements as well. Wouldn't that be a bug in the validator? These are supposed to be what the runtime would add anyway, only ... rendered on the server?

Right. This seems like something the validator should exclude counting any `width` or `height` properties from an inline `style` which also has the `i-amphtml-layout` attribute set. I've filed an issue to exempt `style` attributes added for `[i-amphtml-layout]` elements: https://github.com/ampproject/amphtml/issues/30726.
